zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/instance/DbJobState.java

```diff
@@ -430,7 +430,7 @@ public long findBackedOffJobs(final long timestamp, final BiPredicate<Long, JobR
   boolean visitJob(final long jobKey, final BiPredicate<Long, JobRecord> callback) {
     final JobRecord job = getJob(jobKey);
     if (job == null) {
-      LOG.error("Expected to find job with key {}, but no job found", jobKey);
+      LOG.warn("Expected to find job with key {}, but no job found", jobKey);
```

Comment:
💭  would also be worth a separate commit as both changes are related by also independent